### PR TITLE
Supporting train&val on Maxcompute for TF

### PR DESCRIPTION
**Pull request by weiguoz**
A part of [Train and val](https://github.com/sql-machine-learning/sqlflow/issues/624).

### CODE REVIEW

File: sql/codegen.go [hunk 1]
```diff
@@ -248,309 +261,17 @@ func genTF(w io.Writer, pr *extendedSelect, ds *trainAndValDataset, fts fieldTyp
 	if e != nil {
 		return e
 	}
-	// TODO(weiguo): fix codegen to carry trainingDatase in the next PR
-	if e = codegenTemplate.Execute(w, r); e != nil {
-		return fmt.Errorf("genTF: failed executing template: %v", e)
+	if pr.train {
```
Comment (tonyyang-svail) [inline sql/codegen.go:264]: How about
```go
if pr.train {
    return tfTrainTemplate.Execute(w, r)
}
return tfPredTemplate.Execute(w, r)
```

File: sql/template_tf.go [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+feature_metas["{{$value.FeatureName}}"] = {
+    "feature_name": "{{$value.FeatureName}}",
+    "dtype": "{{$value.Dtype}}",
+    "delimiter": "{{$value.Delimiter}}",
+    "shape": {{$value.InputShape}},
+    "is_sparse": "{{$value.IsSparse}}" == "true"
+}
+{{end}}
+
+def get_dtype(type_str):
```
Comment (tonyyang-svail) [inline sql/template_tf.go:96]: shall we move `get_dtype`and `_parse_sparse_feature` to `sql/python` so that they be shared between templates?

**Review by tonyyang-svail — DISMISSED** [review]
Generally LGTM.

File: sql/template_tf.go (hunk 2 above)
Comment (weiguoz) [reply sql/template_tf.go:96]: `get_dtype` & `_parse_sparse_feature` are specified to `tensorflow`. I'm not sure if other templates would use this.

**Review by tonyyang-svail — APPROVED** [review]

**Pull request merged**